llvm/lib/Transforms/IPO/LowerTypeTests.cpp

```diff
@@ -1494,6 +1494,9 @@ void LowerTypeTestsModule::replaceWeakDeclarationWithJumpTablePtr(
                                      Constant::getNullValue(F->getType()));
     Value *Select = Builder.CreateSelect(ICmp, JT,
                                          Constant::getNullValue(F->getType()));
+    
+    if (auto *SI = dyn_cast<SelectInst>(Select))
+      setExplicitlyUnknownBranchWeightsIfProfiled(*SI, DEBUG_TYPE);
```

Comment:
@pcc a question - would it be technically possible to propagate information about the weak function from [thin]link over to this point, and thus know if we even need a `select` in the first place?